server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -227,7 +227,7 @@ public void registerShuffle(
             taskInfo.refreshLatestStageAttemptNumber(shuffleId, stageAttemptNumber);
             try {
               long start = System.currentTimeMillis();
-              shuffleServer.getShuffleTaskManager().removeShuffleDataSync(appId, shuffleId);
+              shuffleServer.getShuffleTaskManager().softRemoveShuffleDataSync(appId, shuffleId);
```

Comment:
I hope this could be enabled by the extra config option

Reply:
Sorry for my previous comment, I think this deletion could be named as TwoPhaseDeletion, which will include 2 phases
1. Soft deletion
2. Hard deletion

And for the original  deletion way is the hard deletion, we could extra the abstract class to have a good abstraction

Reply:
Could you use another concept? Because hard deletion makes me think of `rm --force`. Maybe you can use `rename` directly.